### ISSUE TITLE
[Stepper] Ensure text and images of text provide sufficient contrast (CAL-AMP32) 

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#13536]
**monday.com sync:** #10821959468

### Violation: 
 Ensure text and images of text provide sufficient contrast 

### WCAG Reference: 
- [1.4.3 Contrast (Minimum)](https://www.w3.org/WAI/WCAG22/Understanding/contrast-minimum.html) 

### Recommended solution
_TBD_, design consultation needed to determine how to provide sufficient contrast to the component's numbers and content (e.g., "1. Add info")

<img width="951" height="210" alt="Image" src="https://github.com/user-attachments/assets/7c2c4829-0af9-471f-b0e5-ba6fcd5738a4" />

### Acceptance Criteria 

- [ ] Update the color token of the selected state number affordance from `color.brand` to `color.text.1`
- [ ] Update the color token of the error state number affordance from `color.danger.default` to `color.text.1`

[Link to full Figma Specs 🔗](https://www.figma.com/design/UwR70KEEnoZ5J2KMKSKCA0/-Stepper--Ensure-text-and-images-of-text-provide-sufficient-contrast?node-id=455-4748&t=d3O7lORhcRaMbNvX-11)

<img width="2480" height="2548" alt="Image" src="https://github.com/user-attachments/assets/775a599c-9580-468d-b16a-681db09bfed1" />

### Severity: 
 Low 

### Media Type: 
 Color and Contrast 

--- 
### Areas for Remediation: 
- Title: CAL-AMP32 Ensure text and images of text provide sufficient contrast 
- Module: M5 Stepper, Tabs, Link, FAB 
- Finding ID: [CAL-AMP32](https://esri.hub.essentia11y.com/short-link/wwjs1xOaGd4RdPKz) 

**Issue**
There is text that does not meet the required minimum color contrast ratio with its background.

Foreground color: #007AC2
Background color: #F7F7F7
Contrast ratio: 4.30:1
Required contrast ratio for this content: 4.50:1

Examples of text with this contrast ratio include:
- "1." (in selected state)
- "2."
- "3."
- "4.".

**User Impact**
Users with low vision will have difficulty reading this content.

**Code Reference** 
 ```
<div class="stepper-item-number"><!--?lit$973891438$-->1.</div>

```

**Steps to Reproduce**
1. Locate the text.
2. Inspect it with Chrome DevTools.
3. In the Styles tab, copy the text color and background color.
4. Use a calculator such as https://www.levelaccess.com/color-contrast-checker-new/ to determine the contrast ratio.
5. Notice that the contrast ratio is below the required value for this size of text.
 
**Recommendation**
Ensure text and images of text provide sufficient contrast. The following contrast ratios are required:

- Text smaller than 18 pt (24 px), or smaller than 14 pt (19 px) if bold, must have a color contrast ratio of 4.50:1 or more with adjacent colors.

- Text 18 pt (24 px) or larger, or 14 pt (19 px) or larger if bold, must have a color contrast ratio of 3.00:1 or more with adjacent colors.

Disabled controls that do not accept user interaction are exempt from this requirement.

To calculate color contrast ratios, use a tool such as the Level Access Accessible Color Picker Chrome extension: https://chrome.google.com/webstore/detail/accessible-color-picker/bgfhbflmeekopanooidljpnmnljdihld or the Color Contrast Checker: https://www.levelaccess.com/color-contrast-checker-new/

**Compliant Code Example**
```
<!-- text with a contrast ratio of 12.6:1 with the background color -->
<style>
 body {
 background-color: #FFFFFF;
 color: #333333;
 }
</style>

<h1>Welcome to our Website</h1>
<p>This is an example of non-compliant text with insufficient color contrast.</p>
```

 **Issue Image** 
 https://desktopqa/aic_lap/Calcite/Calcite2025/CAL-AMP32.png 
 

 --- 
### Additional Resources: 
- [LA Platform Report Link](https://esri.hub.essentia11y.com/short-link/wwjs1xOaGd4RdPKz) 
- [Esri Accessibility Resources](https://esriis.sharepoint.com/sites/Accessibility)
- Report Source: Product accessibility evaluation conducted on Calcite by Level Access and delivered August, 2025.

**Comment by stevenalbers** [comment]
Semi-related issue to this, there's 0 contrast on high contrast Windows themes. Here's `complete`, `selected`, `error`, `disabled` with no contrast themes, and with Night Sky:

<img width="397" height="48" alt="Image" src="https://github.com/user-attachments/assets/cd73fb52-ffbf-462e-be8a-3dc93f065fee" />
<img width="395" height="47" alt="Image" src="https://github.com/user-attachments/assets/0b262d64-061b-4fab-86d5-e6af02ecd30b" />

Windows 11
Settings -> Accessibility -> Contrast themes -> Night Sky (Applies to all themes)
calcite-components@5.2.0-next.70

**Comment by stevenalbers** [comment]
Did a little more research, turns out this is because the stepper bars use `border-color` to set colors, but (at least in Windows' high contrast themes) border colors are ignored in favor of the one theme-defined border color. In my component I'm working around this by setting the bar css vars on the `forced-colors` media query to use system-defined high contrast colors, but this is something that will likely have to be fixed at the library level as it affects all instances of `calcite-stepper`. And if there's a calcite-standard list of high contrast colors I'd rather defer to that than roll out my own.
